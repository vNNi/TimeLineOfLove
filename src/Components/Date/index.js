import React from 'react';
import {css} from 'glamor';

export default function index(props) {
    const img = css({
        width: '100%',
        height: '40%',
    });

    const title = css({
        textAlign: 'center',
    });

    const font = css({
        fontFamily: 'Muli, sans-serif',
    });

    const desc = css({
        fontSize: '20px',
    });

    return (
        <div>
            <h1 {...title} {...font} >{props.title}</h1>
            <div style={{margin: '0 8%'}}>
                {props.img? <img {...img} src={props.img}/>:null}
            </div>
            <div style={{margin: '0 5%'}}>
                <p {...desc} {...font}>{props.desc}</p>
            </div>
        </div>
    )
}
