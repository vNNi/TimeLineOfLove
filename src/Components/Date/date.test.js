import React from 'react';
import Date from './index';
import renderer from 'react-test-renderer';
import {img} from '../../__mock__/fileMock';
it('date renders correctly without image', () => {
    const tree = renderer.create(<Date title='teste' desc='teste2'/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('date renders correctly with all props',() => {
    const tree = renderer.create(<Date title='teste' desc='teste2' img={img}/>).toJSON();
    expect(tree).toMatchSnapshot();
})