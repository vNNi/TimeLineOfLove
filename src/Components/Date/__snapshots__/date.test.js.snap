// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date renders correctly with all props 1`] = `
<div>
  <h1
    data-css-1yv2jt6=""
    data-css-8pban5=""
  >
    teste
  </h1>
  <div
    style={
      Object {
        "margin": "0 10%",
      }
    }
  />
  <hr />
  <div
    style={
      Object {
        "margin": "0 5%",
      }
    }
  >
    <p
      data-css-8pban5=""
    >
      teste2
    </p>
  </div>
</div>
`;

exports[`date renders correctly without image 1`] = `
<div>
  <h1
    data-css-1yv2jt6=""
    data-css-8pban5=""
  >
    teste
  </h1>
  <div
    style={
      Object {
        "margin": "0 10%",
      }
    }
  />
  <hr />
  <div
    style={
      Object {
        "margin": "0 5%",
      }
    }
  >
    <p
      data-css-8pban5=""
    >
      teste2
    </p>
  </div>
</div>
`;
